charts/incubator/plausible/templates/_secret.tpl

```diff
@@ -0,0 +1,28 @@
+{{- define "plausible.secret" -}}
+enabled: true
+stringData:
+  {{- with (lookup "v1" "Secret" .Release.Namespace "plausible-secret") }}
+  SECRET_KEY_BASE: {{ index .data "SECRET_KEY_BASE" | b64dec }}
+  {{- else }}
+  {{- /* The plain value of SECRET_KEY_BASE is also base64 encoded */}}
+  SECRET_KEY_BASE: {{ randAlphaNum 65 | b64enc }}
+  {{- end }}
```

Comment:
```suggestion
{{- /* The plain value of SECRET_KEY_BASE is also base64 encoded */}}
{{- $key := randAlphaNum 65 | b64enc -}}
{{- with (lookup "v1" "Secret" .Release.Namespace "plausible-secret") -}}
  {{- $key = index .data "SECRET_KEY_BASE" | b64dec -}}
{{- end }}
enabled: true
data:
  SECRET_KEY_BASE: {{ $key }}
```

Personally I like this "better" but I'm fine either way.

Only "nit", it's `data` not `stringData` Even tho content is added as stringData..
(It should have been something like `content` but it's too late now)

Reply:
I found both versions throughout the code base, so I stuck with the one I liked more.

> Just to be sure, the length is correctly 65 right? (I usually see those being multiples of 2).

Not sure how I came up with a length `65` last time. 😅 Plausible recommends using the default [`mix phx.gen.secret`](https://hexdocs.pm/phoenix/Mix.Tasks.Phx.Gen.Secret.html#content) of the Phoenix Framework or `openssl rand -hex 64` to generate the secret.

---

Just because I find it a fun exercise:

`openssl rand -hex 64` generates a 128 character hex string, thus a secret with _512 bits_ of entropy.

[To find the length, _L_, needed to achieve a desired strength _H_, with a password drawn randomly from a set of _N_ symbols, one computes](https://en.wikipedia.org/wiki/Password_strength#Entropy_as_a_measure_of_password_strength):

$$ L = \lceil \dfrac{H}{log_{2} N} \rceil $$

To get a secret with strength _H = 512_ with Helm's `randAlphaNum` function where N = 62:

$$ L = \lceil \dfrac{512}{log_{2} 62} \rceil = 86 $$

So let's go with 86 isntead!